jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -140,8 +141,18 @@ def vue_unit_conversion(self, *args, **kwargs):
                 self.hub.broadcast(snackbar_message)
 
                 return
+
+        # Uncertainty converted to new flux units
+        temp_uncertainty = self.spectrum.uncertainty.quantity.to(u.Unit(set_flux_unit.unit),
+                                                      equivalencies=u.spectral_density(set_spectral_axis_unit))
+
         # Create new spectrum with new units.
-        converted_spec = self.spectrum._copy(flux=set_flux_unit, spectral_axis=set_spectral_axis_unit, unit=set_flux_unit.unit)
+        converted_spec = self.spectrum._copy(flux=set_flux_unit,
+                                             spectral_axis=set_spectral_axis_unit,
+                                             unit=set_flux_unit.unit,
+                                             uncertainty=self.return_uncertainty_type(self.spectrum.uncertainty.uncertainty_type,
+                                                                                      temp_uncertainty.value)
+                                             )
```

Comment:
```suggestion
                                             uncertainty=self.spectrum.uncertainty.__class__(temp_uncertainty.value))
```

I don't think you need to create a method for dealing with uncertainty types, instead just use the class of the input uncertainties. This has the benefit of seamlessly handling any new types of uncertainties the user is using that aren't specified in your `if`/`else` statement.

Reply:
I was looking for some way to do this but couldn't find anything in the documentation, thanks!